pkg/executor/show_stats.go

```diff
@@ -37,19 +38,29 @@ import (
 func (e *ShowExec) fetchShowStatsExtended(ctx context.Context) error {
 	do := domain.GetDomain(e.Ctx())
 	h := do.StatsHandle()
-	dbs := do.InfoSchema().AllSchemaNames()
-	for _, db := range dbs {
-		tables, err := do.InfoSchema().SchemaTableInfos(ctx, db)
+	is := do.InfoSchema()
+	partitionTables := is.ListTablesWithSpecialAttribute(infoschema.PartitionAttribute)
+	partitionTableIDMap := make(map[int64]struct{}, len(partitionTables))
+	for _, tbls := range partitionTables {
+		for _, tbl := range tbls.TableInfos {
+			partitionTableIDMap[tbl.ID] = struct{}{}
+		}
+	}
+
+	for _, db := range is.AllSchemaNames() {
+		simpleTblInfos, err := is.SchemaSimpleTableInfos(ctx, db)
```

Comment:
Get simple table infos，then exclude partition tables，and then get tableInfo one by one